script/verdaccio.js

```diff
@@ -0,0 +1,277 @@
+  
+  generateVerdaccioConfigFile();
+  
+  // Start verdaccio application
+  try {
+    startVerdaccioApp();
+  } catch (e) {
+    logger.error('Failed to start verdaccio', e);
+  }
+  
+  // Make SDK test directory
+  if (!fs.existsSync(sdkTestDir)) {
+    logger.log(`Creating SDK testing directory: ${sdkTestDir}`);
+    fs.mkdirSync(sdkTestDir);
+  } else {
+    logger.log(`SDK testing directory already exists: ${sdkTestDir}`);
+  }
+
+  logger.log(`Navigate to SDK testing directory: ${sdkTestDir}`);
+  process.chdir(sdkTestDir);
+  
+  // Clone SDK from github
+  logger.log(`Cloning ${SDK_APP_NAME} into ${sdkTestDir}`);
+  
+  //Clean up
+  spawnOrFail('rm', ['-rf', `${SDK_APP_NAME}`]);
+  spawnOrFail('git', [
+    'clone',
+    'git@github.com:aws/amazon-chime-sdk-js.git'
+  ]);
+  // navigate in to cloned SDK
+  process.chdir(path.join(`./${SDK_APP_NAME}`));
+  
+  // Update package.json version
+  let package = fs.readFileSync('package.json', 'utf-8');
+
+  const packageJson = JSON.parse(package);
+  const originalVersion = packageJson.version;
+  const versions = originalVersion.split('.').map(x => parseInt(x));
+  versions[2] += 1;
+  const newVersion = versions.join('.');
+
+  logger.log(`Changing version number from ${originalVersion} to ${newVersion}`);
+
+  spawnOrFail('npm', [`version`,`${newVersion}`, '--no-git-tag-version']);
+
+  package = fs.readFileSync('package.json', 'utf-8');
+
+  logger.log(`Temporarily removing prepublishOnly and postpublish`);
+  fs.writeFileSync('package.json', package.replace(`"postpublish": "script/postpublish",`, ""));
+  package = fs.readFileSync('package.json', 'utf-8');
+  fs.writeFileSync('package.json', package.replace(`"prepublishOnly": "script/publish",`, ""));
+  
+  logger.log('Installing dependencies...');
+  spawnOrFail('npm', ['install']);
+  logger.log('Building...');
+  spawnOrFail('npm', ['run', 'build']);
+
+  try {
+    // Cleanup package before publishing new
+    logger.log('Unpublishing any existing packages from verdaccio.');
+    cp.execSync(`npm unpublish --registry ${verdaccioEndpoint} --force`);
+  } catch {
+    logger.log('No existing npm module in verdaccio.');
+  }
+  logger.log('Publishing package to verdaccio.');
+  spawnOrFail('npm', [`publish`, '--registry', `${verdaccioEndpoint}`]);
+  
+  process.chdir('./demos/browser');
+
+  spawnOrFail('npm', [`uninstall ${SDK_APP_NAME}`]);
+
+  //  Install SDK components package from local repo.
+  logger.log(`Installing ${SDK_APP_NAME} from local verdaccio server...`);
+  spawnOrFail('npm', [`install ${SDK_APP_NAME} --registry ${verdaccioEndpoint}`]);
+
+  let meetingV2File = fs.readFileSync('app/meetingV2/meetingV2.ts', 'utf-8');
+  fs.writeFileSync('app/meetingV2/meetingV2.ts', meetingV2File.replace('../../../../src/index', 'amazon-chime-sdk-js'));
+
+  logger.log(`Installing...`);
+  cp.execSync('npm install');
+  cp.execSync('npm install aws-sdk');
+  logger.log(`Building...`);
+  cp.execSync('npm run build');
```

Comment:
Why dont we just exe 'npm run start' here?

Reply:
Two things:
1. Npm run start runs `npm run deps`, which builds the local module, which we dont need to do since we're using the sdk module from verdaccio
2. There was some issues with spawning/closing the processes created with doing exec(npm run start) because npm creates child processes for each npm command inside of npm run start, so I was running into some issues cleaning it up and actually starting the server.